tests/integration/app-charm/src/charm.py

```diff
@@ -74,21 +88,241 @@ def _on_start(self, _) -> None:
 
     def on_topic_created_consumer(self, event: TopicCreatedEvent) -> None:
         logger.info(f"{event.username} {event.password} {event.bootstrap_server} {event.tls}")
-        peer_relation = self.model.get_relation("cluster")
-        peer_relation.data[self.app]["username"] = event.username or ""
-        peer_relation.data[self.app]["password"] = event.password or ""
+        self.peer_relation.data[self.app]["username"] = event.username or ""
+        self.peer_relation.data[self.app]["password"] = event.password or ""
+        self.peer_relation.data[self.app]["bootstrap-server"] = event.bootstrap_server or ""
         return
 
     def on_topic_created_producer(self, event: TopicCreatedEvent) -> None:
         logger.info(f"{event.username} {event.password} {event.bootstrap_server} {event.tls}")
         self.peer_relation.data[self.app]["username"] = event.username or ""
         self.peer_relation.data[self.app]["password"] = event.password or ""
+        self.peer_relation.data[self.app]["bootstrap-server"] = event.bootstrap_server or ""
         return
 
     def on_topic_created_admin(self, event: TopicCreatedEvent) -> None:
         logger.info(f"{event.username} {event.password} {event.bootstrap_server} {event.tls}")
         self.peer_relation.data[self.app]["username"] = event.username or ""
         self.peer_relation.data[self.app]["password"] = event.password or ""
+        self.peer_relation.data[self.app]["bootstrap-server"] = event.bootstrap_server or ""
+        return
+
+    def _install_packages(self):
+        subprocess.check_output(
+            "apt update && DEBIAN_FRONTEND=noninteractive apt install openjdk-21-jre-headless -y -qq",
```

Comment:
**niptick** I would just check that the jdk is consistent with what we are using elsewhere to avoid strange/changes in the behaviour

Reply:
I don't think that's necessary. Bear in mind this is only for a 'int-test' charm, and we're only doing this to install the JDK.
If we want, we can pretend that this is mimicking a 'modern' Java client trying to connect externally to Kafka.